src/sharing/SharingDialog.component.js

```diff
@@ -1,44 +1,251 @@
+
+        return apiInstance.get('sharing/search', { key: searchText })
+            .then((searchResult) => {
+                let transformedResult = searchResult.users.map(
+                    user => transformAccessObject(user, 'user'));
+
+                transformedResult = transformedResult.concat(
+                    searchResult.userGroups.map(
+                        userGroup => transformAccessObject(userGroup, 'userGroup')));
+
+                return transformedResult;
+            });
+    }
+
+    onSharingChanged(updatedAttributes, onSuccess) {
+        const objectToShare = {
+            ...this.state.objectToShare,
+            ...updatedAttributes,
+        };
+
+        const apiObject = this.restoreObjectStructure(objectToShare);
+
+        return this.state.api.post(`sharing?type=${this.props.type}&id=${this.props.id}`, apiObject)
+          .then(({ httpStatus, message }) => {
+              if (httpStatus === 'OK') {
+                  this.setState({
+                      objectToShare,
+                      apiObject,
+                  }, () => {
+                      onSuccess && onSuccess();
+                  });
+              } else {
+                  console.warn('Failed to post changes.');
+                  console.warn('SERVER SAID:', message);
+              }
 
-    mixins: [Translate],
+              return message;
+          })
+          .catch(({ message }) => {
+              console.warn('Failed to post changes.');
+              console.warn('SERVER SAID:', message);
+          });
+    }
+
+    loadObjectFromApi() {
+        getInstance().then((d2) => {
+            const apiInstance = d2.Api.getApi();
+            apiInstance.get('sharing', { type: this.props.type, id: this.props.id })
+                .then((apiObject) => {
+                    this.setState({
+                        api: apiInstance,
+                        apiObject,
+                        objectToShare: transformObjectStructure(apiObject.meta, apiObject.object),
+                    });
+                });
+        });
+    }
+
+    restoreObjectStructure(transformedObject) {
+        const userAccesses = [];
+        const userGroupAccesses = [];
+
+        // eslint-disable-next-line array-callback-return
+        transformedObject.accesses.map((access) => {
+            const apiAccess = {
+                id: access.id,
+                name: access.name,
+                displayName: access.name,
+                access: cachedAccessTypeToString(access.canView, access.canEdit),
+            };
+
+            access.type === 'user'
+                ? userAccesses.push(apiAccess)
+                : userGroupAccesses.push(apiAccess);
+        });
+
+        return {
+            meta: this.state.apiObject.meta,
+            object: {
+                ...this.state.apiObject.object,
+                ...userAccesses && { userAccesses },
+                ...userGroupAccesses && { userGroupAccesses },
```

Comment:
This could be written as just `userAccesses` and `userGroupAccesses`? The variable will always be initialized as you do that on lines 148 and 149. 

```js
object: {
    ...this.state.apiObject.object,
    userAccesses,
    userGroupAccesses,
```